portfolio/src/main/java/com/google/sps/data/ReactionDatastore.java

```diff
@@ -0,0 +1,101 @@
+package com.google.sps.data;
+
+import static com.google.sps.data.Reaction.LAUGH;
+import static com.google.sps.data.Reaction.LIKE;
+import static com.google.sps.data.Reaction.LOVE;
+import static com.google.sps.data.Reaction.NONE;
+import static com.google.sps.data.Reaction.WOW;
+import static com.google.sps.utils.ReactionUtils.toPropertyName;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Transaction;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ReactionDatastore {
+
+  private static ReactionDatastore instance = new ReactionDatastore();
+
+  private DatastoreService datastore;
+
+  private ReactionDatastore() {
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  public static ReactionDatastore getInstance() {
+    return instance;
+  }
+
+  public Post getReactions(String postId) {
+    try {
+      Entity postEntity = datastore.get(KeyFactory.createKey("Post", postId));
+
+      Post post = new Post(postId);
+      post.setLikeCount((Long) postEntity.getProperty("likeCount"));
```

Comment:
Just curious, any reason we are using wrapper class (Long) instead of long here?

Reply:
Was initially using a Map to store values (which requires the wrapper class) so this is an artefact of that implementation. I am switching back to using a Map for flexibility but I will address this as even then it is not strictly necessary, it should get up-casted without any issues.